src/protocols/ethereum/codegen/abi.js

```diff
@@ -208,6 +225,54 @@ module.exports = class AbiCodeGenerator {
             `return new ${paramsClassName}(this)`,
           ),
         )
+
+        // Fixture generation
+        try {
+          const args = yaml.parse(fs.readFileSync('./fixtures.yaml', 'utf8'))
+          const blockNumber = args['blockNumber']
+          const contractAddr = args['contractAddr']
+          const topic0 = args['topic0']
+          const apiKey = args['apiKey']
+          const url = `https://api.etherscan.io/api?module=logs&action=getLogs&fromBlock=${blockNumber}&toBlock=${blockNumber}&address=${contractAddr}&${topic0}=topic0&apikey=${apiKey}`;
+
+          let resp = request("GET", url)
+          let body = JSON.parse(resp.getBody("utf8"))
+          if (body.status === '0') {
+            throw new Error(body.result)
+          }
+
+          let res = web3.eth.abi.decodeLog(
+            namesAndTypes,
+            body.result[0].data,
+            []
+          );
+
+          let stmnts = ""
+          for (let i = 0; i < namesAndTypes.length; i++) {
+            let code = '"' + res[i] + '"'
+            if (namesAndTypes[i].type.toString() == "address") {
+              code = `Address.fromString(${code})`
+            }
+            stmnts = stmnts.concat(`event.parameters.push(new ethereum.EventParam(\"${namesAndTypes[i].name}\", ${typesCodegen.ethereumFromAsc(code, namesAndTypes[i].type)}));`, `\n`)
+          }
+
+          klass.addMethod(
+            tsCodegen.staticMethod(
+              `mock${eventClassName}`,
+              [],
+              tsCodegen.namedType(eventClassName),
+              `
+              let event = changetype<${eventClassName}>(newMockEvent());
+              ${stmnts}
+              return event;
+              `,
+            )
+          )
+        } catch (e) {
+            // no fixtures
+            console.log(e)
```

Comment:
Also I'm not sure we want to log the errors for every time, since a lot of the times the fixture file won't be there.

Reply:
This could also be made conditionally.